vxfreeswitch/client.py

```diff
@@ -0,0 +1,128 @@
+# -*- test-case-name: vxfreeswitch.tests.test_client -*-
+
+"""
+FreeSwitch ESL API client.
+"""
+
+from twisted.internet.protocol import ClientFactory
+from twisted.internet.defer import inlineCallbacks, returnValue, Deferred
+
+from eventsocket import EventProtocol, EventError
+
+
+class FreeSwitchClientProtocol(EventProtocol):
+    """ Freeswitch ESL client.
+
+    :param str auth:
+        Authentication string to send to FreeSwitch.
+    """
+
+    def __init__(self, auth):
+        EventProtocol.__init__(self)
+        self._auth = auth
+        self._connected = Deferred()
+
+    @inlineCallbacks
+    def connectionMade(self):
+        if self._auth:
+            yield self.auth(self._auth)
```

Comment:
I found that in practice ``authRequest`` isn't called, but I would like to understand more clearly why that is.